libsolidity/ast/AST.cpp

```diff
@@ -530,20 +530,26 @@ IdentifierAnnotation& Identifier::annotation() const
 	return dynamic_cast<IdentifierAnnotation&>(*m_annotation);
 }
 
+bool Literal::hasHexPrefix() const
+{
+	if (token() != Token::Number)
+		return false;
+	return value().substr(0, 2) == "0x";
```

Comment:
I rather like the `boost:starts_with()` alternative because it doesn't allocate a new string object, but it's a nitpick.